dps_client/src/dps_sasl_tpm.c

```diff
@@ -124,7 +124,7 @@ static int construct_send_data(SASL_TPM_INSTANCE* sasl_tpm_info, unsigned char*
     }
     else
     {
-        payload_data_length = data_len + 1;
```

Comment:
>data_len + 1 [](start = 30, length = 12)

Are we concerned about overflow bugs here?  If we are, the cast complicates things.

Reply:
On 64 bit system the compiler requires a cast.

---
In reply to: [138971318](https://github.com/Azure/azure-iot-sdk-c/pull/231#discussion_r138971318) [](ancestors = 138971318)